packages/clarity-js/src/data/metadata.ts

```diff
@@ -120,14 +131,22 @@ function userAgentData(): void {
 export function stop(): void {
     rootDomain = null;
     data = null;
+    consentStatus = null;
     for (const cb of callbacks) {
-        cb.called = false;
+        cb.called = cb.consentInfo;
     }
 }
 
-export function metadata(cb: MetadataCallback, wait = true, recall = false): void {
+export function metadata(cb: MetadataCallback, wait = true, recall = false, consentInfo = false): void {
     const upgraded = config.lean ? BooleanFlag.False : BooleanFlag.True;
     let called = false;
+  
+    if (consentInfo) {
+      data.consent = consentStatus
+    } else{
+        delete data.consent;
```

Comment:
you will have multiple callbacks, some with consentInfo and some without - this will not work. as you return you might want to add consentstatus instead of adding to data